freeipa/src/main/java/com/sequenceiq/freeipa/flow/stack/start/OneFreeIpaReachableAttempt.java

```diff
@@ -0,0 +1,62 @@
+package com.sequenceiq.freeipa.flow.stack.start;
+
+import static com.sequenceiq.cloudbreak.util.Benchmark.checkedMeasure;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.dyngr.core.AttemptMaker;
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.sequenceiq.cloudbreak.cloud.scheduler.PollGroup;
+import com.sequenceiq.cloudbreak.cloud.store.InMemoryStateStore;
+import com.sequenceiq.freeipa.client.model.RPCResponse;
+import com.sequenceiq.freeipa.entity.InstanceMetaData;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.stack.FreeIpaHealthDetailsService;
+
+public class OneFreeIpaReachableAttempt implements AttemptMaker<Void> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(OneFreeIpaReachableAttempt.class);
+
+    private final FreeIpaHealthDetailsService freeIpaHealthDetailsService;
+
+    private final Stack stack;
+
+    private final Set<InstanceMetaData> instanceMetaDataSet;
+
+    public OneFreeIpaReachableAttempt(FreeIpaHealthDetailsService freeIpaHealthDetailsService, Stack stack, Set<InstanceMetaData> instanceMetaDataSet) {
+        this.freeIpaHealthDetailsService = freeIpaHealthDetailsService;
+        this.stack = stack;
+        this.instanceMetaDataSet = instanceMetaDataSet;
+    }
+
+    @Override
+    public AttemptResult<Void> process() {
+        if (stack == null || instanceMetaDataSet == null || instanceMetaDataSet.isEmpty()) {
```

Comment:
You can check this in the constructor or in the factory.

Reply:
OK, thanks